packages/kbn-securitysolution-autocomplete/src/field_value_wildcard/index.tsx

```diff
@@ -0,0 +1,245 @@
+
+// TODO: I have to use any here for now, but once this is available below, we should use the correct types, https://github.com/elastic/kibana/issues/100715
+// import { AutocompleteStart } from '../../../../../../../src/plugins/data/public';
+type AutocompleteStart = any;
+
+import * as i18n from '../translations';
+import { useFieldValueAutocomplete } from '../hooks/use_field_value_autocomplete';
+import {
+  getGenericComboBoxProps,
+  GetGenericComboBoxPropsReturn,
+} from '../get_generic_combo_box_props';
+import { paramIsValid } from '../param_is_valid';
+
+const SINGLE_SELECTION = { asPlainText: true };
+
+interface AutocompleteFieldWildcardProps {
+  placeholder: string;
+  selectedField: DataViewFieldBase | undefined;
+  selectedValue: string | undefined;
+  indexPattern: DataViewBase | undefined;
+  isLoading: boolean;
+  isDisabled: boolean;
+  isClearable: boolean;
+  isRequired?: boolean;
+  fieldInputWidth?: number;
+  rowLabel?: string;
+  autocompleteService: AutocompleteStart;
+  onChange: (arg: string) => void;
+  onError: (arg: boolean) => void;
+  onWarning: (arg: boolean) => void;
+  warning?: string;
+}
+
+export const AutocompleteFieldWildcardComponent: React.FC<AutocompleteFieldWildcardProps> = ({
+  autocompleteService,
+  placeholder,
+  rowLabel,
+  selectedField,
+  selectedValue,
+  indexPattern,
+  isLoading,
+  isDisabled = false,
+  isClearable = false,
+  isRequired = false,
+  fieldInputWidth,
+  onChange,
+  onError,
+  onWarning,
+  warning,
+}): JSX.Element => {
+  const [searchQuery, setSearchQuery] = useState('');
+  const [touched, setIsTouched] = useState(false);
+  const [error, setError] = useState<string | undefined>(undefined);
+  const [isLoadingSuggestions, , suggestions] = useFieldValueAutocomplete({
+    autocompleteService,
+    fieldValue: selectedValue,
+    indexPattern,
+    operatorType: OperatorTypeEnum.WILDCARD,
+    query: searchQuery,
+    selectedField,
+  });
+  const getLabel = useCallback((option: string): string => option, []);
+  const optionsMemo = useMemo((): string[] => {
+    const valueAsStr = String(selectedValue);
+    return selectedValue != null && selectedValue.trim() !== ''
+      ? uniq([valueAsStr, ...suggestions])
+      : suggestions;
+  }, [suggestions, selectedValue]);
+  const selectedOptionsMemo = useMemo((): string[] => {
+    const valueAsStr = String(selectedValue);
+    return selectedValue ? [valueAsStr] : [];
+  }, [selectedValue]);
+
+  const handleError = useCallback(
+    (err: string | undefined): void => {
+      setError((existingErr): string | undefined => {
+        const oldErr = existingErr != null;
+        const newErr = err != null;
+        if (oldErr !== newErr && onError != null) {
+          onError(newErr);
+        }
+
+        return err;
+      });
+    },
+    [setError, onError]
+  );
+
+  const handleWarning = useCallback(
+    (warn: string | undefined): void => {
+      onWarning(warn !== undefined);
```

Comment:
opinion/no changes requested, but just how I do things...

You can be strict with the `!== undefined` if you want, but typically I just do `!= null` everywhere unless I am needing to distinguish between `undefined` and `null` which I try to avoid having to distinguish. I usually do it just in case someone later changes it to `null`. As some programmers want to always use `null` for values and others `undefined` ... 🤷 . Just letting you know. ;-)